### ISSUE TITLE
New default configuration for iCade is buggy

### ISSUE THREAD

**Issue by GoogleCodeExporter** [kexplo/imame4all#30]
```
What steps will reproduce the problem?
1. Start any game that utilizes six buttons using iCade
2. Select (lower red button) and black button (top left button) are switched on 
iCade in new default configuration
3. Havoc ensues

What is the expected output? What do you see instead?

I liked the two red buttons as the nonaction buttons.  The new configuration 
has led to odd effects, such as
Not being able to use some buttons at the same time as the directional stick.  
To see what I mean, try reconfiguring the two top black buttons as the leftband 
right punch buttons in Punchout.  You cannot use the left punch button at the 
same time as holding the stick in the up position.

What version of the product are you using? On what operating system?

 1.8.1 on iPad 2

Please provide any additional information below.

Please return to the old configuration

```

Original issue reported on code.google.com by `chchawk...@gmail.com` on 17 Aug 2011 at 12:32

**Comment by GoogleCodeExporter** [comment]
```
Coin and start buttons can't be redefined well now and then... you should 
redefine the rest of them l1,l2,a,x,b,y...

the new configuration is related to this thread...

http://boards.openpandora.org/index.php?/topic/4749-icade-mode-button-mapping/

write there if you think another configuration could be better i dont mind 
change the buttons if there a deal.
```

Original comment by `seleuco....@gmail.com` on 17 Aug 2011 at 4:16

**Comment by GoogleCodeExporter** [comment]
```
Coin and start buttons can't be redefined well now and then... you should 
redefine the rest of them l1,l2,a,x,b,y...

the new configuration is related to this thread...

http://boards.openpandora.org/index.php?/topic/4749-icade-mode-button-mapping/

write there if you think another configuration could be better i dont mind 
change the buttons if there a deal.
```

Original comment by `seleuco....@gmail.com` on 17 Aug 2011 at 6:46

**Comment by GoogleCodeExporter** [comment]
```
I'm having the same issue.  I've remapped all the buttons and have the same 
behaviour where the top-left black button doesn't work at the same time as the 
directional stick.  I much prefer the following button configuration:

ST  B1  B2  B3
C1  B4  B5  B6

Will this actually require a change in the code for this to work or is there 
something I'm missing in remapping the buttons?
```

Original comment by `eva...@gmail.com` on 22 Aug 2011 at 2:28

**Comment by GoogleCodeExporter** [comment]
```
as i say before you can remap coin, start... i would try to find a solution in 
the next iOS version but im working now in an android version so i think you 
have to wait a little.
```

Original comment by `seleuco....@gmail.com` on 22 Aug 2011 at 11:49

**Comment by GoogleCodeExporter** [comment]
```
I don't understand what you mean.

Both the OP and myself are remapping start and coin.  The problem is that after 
doing so, you can't play many games since something is causing the top left 
black button to not function at the same time the joystick is moved.  For 
example, when you play a SHMUP you can't move and shoot at the same time.

So simply remapping the buttons doesn't work.
```

Original comment by `eva...@gmail.com` on 23 Aug 2011 at 12:40

**Comment by GoogleCodeExporter** [comment]
```
Dont remap coin and start...  They have some internal mame hooks so it doesn't  
work at all...
```

Original comment by `seleuco....@gmail.com` on 23 Aug 2011 at 4:53

**Comment by GoogleCodeExporter** [comment]
```
Implemented in v1.9.0
```

Original comment by `squid123...@gmail.com` on 22 Sep 2011 at 12:15
- Changed state: **Done**

**Comment by GoogleCodeExporter** [comment]
Original comment by `squid123...@gmail.com` on 22 Sep 2011 at 1:06
- Changed state: **Fixed**
